clientv3/client.go

```diff
@@ -372,12 +375,15 @@ func (c *Client) directDialCreds(ep string) grpccredentials.TransportCredentials
 			// Set the server name must to the endpoint hostname without port since grpc
 			// otherwise attempts to check if x509 cert is valid for the full endpoint
 			// including the scheme and port, which fails.
-			host, _ := endpoint.ParseHostPort(hostPort)
+			host, _, err := net.SplitHostPort(hostPort)
+			if err != nil {
```

Comment:
this returns err if the hostPort had no port... is that what we want? ParseHostPort tolerated that previously

Reply:
Thanks. Fixed.

Reply:
lgtm. Also, looks like the return signature doesn't need to change now.